app/js/trouble.js

```diff
@@ -53,15 +46,15 @@
 			$dropdown.data("status", 'closed');
 			$dropdown.removeClass('gbl_dropdown_active');
 			Array.prototype.foreach = function(cb) {
-				for (i = 0; i < this.length; ++i) {
+				for (var i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			};
+			while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
While this removes the "making" of a function in a loop, the desired change here is to either declare the function outside the loop and pass in the name of the function as the first argument to setTimeout or to assign the anonymous function to a variable outside the loop and pass in the name of the variable. By invoking the code here, it means the function will be called as soon as the code executes rather than in response to the timeout even later.